src/components/Button/Button.tsx

```diff
@@ -54,24 +53,28 @@ const buttonModifierAndStyle = (variant?: VariantType, modifier?: ModifierType)
  * @see https://rivet.uits.iu.edu/components/forms/buttons/#small-buttons
  */
 
-const buttonSize = (size?: SizeType) => (size !== undefined ? `${buttonClass}--${size}` : None);
+const buttonSize = (props: ButtonProps) => 
+  props.size
+  ? `${buttonClass}--${props.size}` 
+  : '';
 
 export const Button: React.SFC<ButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>> = 
-  ({ className, children, id = Rivet.shortuid(), innerRef, modifier, onClick, size, variant, ...attrs}) => {
-  const classes = classNames(buttonModifierAndStyle(variant, modifier), buttonSize(size), className);
+  ({ className, children, id = Rivet.shortuid(), innerRef, onClick, ...props}) => {
```

Comment:
Naming things. I agree `...props` makes more sense than `...attrs`, as that's what they are called in React. My hesitation is that we sometimes use `props` for all the props passed to a component (not destructured). I actually generally use `...others` in my personal projects, but I'm not claiming that's a good name. I'm fine with `...props` over `...attrs`, but thought I'd point out the overload.

Reply:
This should be changed back to `attrs` since that's the standard the rest of the project uses

Reply:
I'll also add that I really prefer spreading the props that this component actually uses.  I understand the workaround you're doing here, but if I was looking at this component to figure out what it did the props line is one of the first places I look.

Reply:
Thanks for the feedback -- I'll take another swing.